dictcc.el

```diff
@@ -271,11 +302,18 @@ At the moment they are of the form `<tr id='trXXX'></tr>'."
                                (dictcc--insert-candidate #'cdr))))))
     (helm :prompt "Filter: " :sources (list source))))
 
+(defun dictcc--select-language (prompt)
+  "Select language with completing read with prompt PROMPT"
+  (let ((lang (completing-read prompt  dictcc-languages-alist)))
+    (cdr (assoc lang dictcc-languages-alist))))
+
 ;;;###autoload
-(defun dictcc (query)
+(defun dictcc (query &optional ask-languages)
   "Search dict.cc for QUERY and insert a result at point."
-  (interactive "sQuery: \n")
-  (dictcc--request query))
+  (interactive "sQuery: \np")
+  (let ((dictcc-source-lang (if (= ask-languages 4) (dictcc--select-language "Source language: ") dictcc-source-lang))
```

Comment:
I am sorry, I got this wrong in my suggestion. With `C-u C-u` we want *both* to be selected, so we need to check `(>= ask-languages 4)` here.

Reply:
That was not your fault, you made the idea very clear and I intended to make it work exactly like that. I just hacked faster than I could think ;-)